### ISSUE TITLE
`lsp-dart-line-length` setting not considered when formatting the buffer.

### ISSUE THREAD

**Issue by laynor** [emacs-lsp/lsp-dart#75]
**Describe the bug**
`lsp-format-buffer` does not respect line length customize settings.

**Note**
Before logging the bug please make sure that the issue is reproducible with latest melpa packages.

**To Reproduce**
`M-x customize-variable lsp-dart-line-length RET`, edit to 120, save
create a flutter project
```
$ flutter create test_project
```
edit `main.dart`
change this line
```
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
```
to this
```
            Text('$_counter', style: Theme.of(context).textTheme.headline4, textDirection: TextDirection.ltr),
```
format the buffer
`M-x lsp-format-buffer`
The line is broken into multiple lines, even if it does not exceed 120 columns.

I enabled lsp io logging to have a look, and I cannot see the `dart.lineLength` setting anywhere in the logs.

**Expected behavior**
The edited line should not be broken into multiple lines
**Screenshots**

**Version**
Include here the result of: <kbd>M-x</kbd> `lsp-dart-version`
[LSP Dart] 1.17.6 (package: 20200922.1550) at 2020.10.12 @ Emacs 28.0.50
[Dart SDK] Dart SDK version: 2.10.1 (stable) (Tue Oct 6 10:54:20 2020 +0200) on "linux_x64"

**Comment by ericdallo** [comment]
Thanks for the report, I'll take a look

**Comment by ericdallo** [comment]
Hey @laynor, it was a bug on `lsp-dart`, could you confirm it was fixed on latest release, `1.17.7`?
Also, this feature only works for Dart SDK version >= 2.10.0

**Comment by laynor** [comment]
@ericdallo Many thanks for the fix, I confirm it's working :+1:

**Comment by ericdallo** [comment]
@laynor maybe the flag changed or something on dart sdk 2.14, could you try with a older version of flutter just to confirm?

**Comment by TheWalkingFridge** [comment]
Happens to me too:
```
Dart SDK version: 2.16.0-10.0.dev
[LSP Dart] - 1.20.0
```

**Comment by ericdallo** [comment]
@TheWalkingFridge this was fixed on 1.20.1

**Comment by martin-dimi** [comment]
@ericdallo updated to 1.21. Can confirm it works :pray:  thanks
